[Yilun Ni] iP

lunzard opened the pull request:
Pull request  required by Week 4 ip

lunzard committed:
Update Level-0

Implement an initial skeletal version of the Duke that simply greets the user and exits.
I add them as seperate methods for easier modification in the future

lunzard committed:
complete Level-1: Greet.Echo.Exit

lunzard committed:
complete Level-2: Add,List

lunzard committed:
complete Level-3: Mark as Done

lunzard committed:
complete Level 4: ToDos, Events,Deadlines

lunzard committed:
attempted Automated Text UI Testing

lunzard committed:
update A-CodeQuality: Improve Code Quality

lunzard committed:
update A-TextUiTesting: Automated Text UI Testing

lunzard commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+
+    public void setBy(String by) {
+        this.by = by;
+    }
```
some of the methods such as `getBy()` and `setBy()` are redundant at this point of time, but I felt that they will be needed in the future. Thus they are kept, which is against the  CodeQuality requirement.

lunzard commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+    private static String command;  // one question: when to make those variable static?
+    private static String description;
+    private static String taskName;
+    private static String taskTime;
```
Not very sure whether too many `static `variables are good for the program, especially in terms of space and effeciency

lunzard commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+                addTask(new Event(taskName,taskTime));
+                break;
+            case "BYE":
+                isQuit = true;
```
seems like the `isQuit` is not included in a method, which might violate `CodeQuality`, but one line of code is too short for extraction, from my point of view

lunzard commented on text-ui-test/runtest.bat:
```diff
@@ -7,15 +7,15 @@ REM delete output from previous run
 
 REM compile the code into the bin folder
-javac  -cp ..\src -Xlint:none -d ..\bin ..\src\main\java\Duke.java
+javac  -cp ..\src\main\java -Xlint:none -d ..\bin ..\src\main\java\*.java
```
not sure whether such setting is correct. I did not see any `successful` or `failure` in my terminal

lunzard reviewed (commented):
Looks Good.
Just need some clarification 
I am confident with my Style of Coding 👍

e0425705 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+    public static void doneTask(int taskIndex){
```
Do you think that there is an indentation error here?

e0425705 replied:
Do you think that there is an indentation error here?

siewwencode commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+                + "Now you have "+ taskCount + " tasks in the list \n"
+                + HORIZONTAL_LINE);
+    }
+    public static void listTasks(){
```
same issue here, minor indentation error

lunzard replied:
hi Siew wen Thanks for the review! I should put a space between my close bracket and open brace.

e0425705 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+public class Deadline extends Task{
```
Do you think that there is an indentation error here?

e0425705 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    @Override
+    public String toString(){
```
Do you think that there is an indentation error here?

e0425705 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
-        System.out.println("Hello from\n" + logo);
+        System.out.println(HORIZONTAL_LINE+ "\n" + logo);
+    }
+    public static void greet(){
```
Do you think that there is an indentation error here?

e0425705 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+        System.out.println(HORIZONTAL_LINE + "\n" + userInput + "\n" + HORIZONTAL_LINE);
+    }*/
+
+    public static void addTask(Task t){
```
Do you think that there is an indentation error here?

e0425705 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static String processCommand(){
```
Do you think that there is an indentation error here?

siewwencode reviewed (commented):
Overall, well done. Other than some minor indentation errors,  i like the way you name the variables and method.

lunzard committed:
Update A-CodeQuality: Improve CodeQuality

lunzard committed:
Update A-TextUiTesting

remove extra spaces as the editor automaticallly trim spaces at the end of each line, now the output.txt and expected.txt are consistent

lunzard committed:
update Level-5: Handle Errors -- create EmptyDescriptionException

MuhammadHoze commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+                + HORIZONTAL_LINE);
+    }
+
+    public static void showLogo() {
```
Nice way of displaying these messages by just calling a function

MuhammadHoze commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,136 @@
+    public static void listTasks() {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Here are the tasks in your list:");
+        for(int i = 0; i< taskCount; i++){
```
Do take note of the "{" spacing after the for loop ()

lunzard replied:
thanks Bro!

MuhammadHoze reviewed (commented):
Nicely done! structure code, rare instances of coding violations.

lunzard committed:
update Level-5: Handle Errors -- exception for done

lunzard committed:
update A-Exceptions -- create new DukeException class

lunzard committed:
Merge branch 'branch-Level-5' into master

lunzard committed:
complete A-Packages: Organize into Packages

lunzard committed:
Merge branch 'branch-A-Packages' into master

lunzard committed:
Level-6: Delete & A-Collections

change the fixed array into flexible ArrayList
Add a delete function

lunzard committed:
Merge branch 'branch-Level-6' into master

lunzard committed:
Level 7: Save

add autosave features by implementing read and writing files.

lunzard committed:
Merge branch 'branch-Level-7' into master

# Conflicts:
#	src/main/java/duke/Duke.java

lunzard committed:
A-Jar

change the seeting so that the program is packaged into a jar file

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+        Scanner in = new Scanner(System.in);
+        boolean isQuit = false;
+
+        //when the program starts
```
This comment may be a little unnecessary.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+    private static String command;  // one question: when to make those variable static?
+    private static String description;
+    private static String taskName;
+    private static String taskTime;
```
These variables are in a way acting like global variables. Is there a way we can avoid this?

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+        System.out.println(HORIZONTAL_LINE + "\n" + greeting + HORIZONTAL_LINE);
+    }
+    public static void bye() {
+        String bye_word = "Bye. Hope to see you again soon!";
```
Variable name should be in camel case and not have underscore.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+                        + "What can I do for you?\n";
+        System.out.println(HORIZONTAL_LINE + "\n" + greeting + HORIZONTAL_LINE);
+    }
+    public static void bye() {
```
Method name should be a verb.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+    }
+
+    public static void addTask(Task t) {
+        tasks.add(t);
```
Although the scope of `t` is small, we can use a better name for it.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static Boolean checkTaskNum() {
```
For boolean methods, your method name should sound boolean.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+            System.out.println(HORIZONTAL_LINE+"\n"
+                    + "☹ OOPS!!! The task number of the one you have done is invalid!\n"
+                    +HORIZONTAL_LINE);
+        }catch (DukeException e) {
```
This catch block is unnecessary, you can just add the logic in the block to the if block above instead of throwing a error there. You should throw error if you want a higher level method to handle it.

jtankw3 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    public boolean isDone() {
+        return this.isDone;
```
You do not need to use the `this` keyword unless it is ambiguous (shadowed by method parameter like in `setName`)

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+
+
+
+    public static String processCommand() {
```
This method may be confusing as it also starts with `process` but unlike the other methods it returns a String instead of being a void method.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+                }
+                break;
+            case "DEADLINE":
+                if(checkDescription() && checkDeadline()) {
```
It may be possible to further apply SLAP here and abstract the details of the checks

jtankw3 reviewed (commented):
Good work! I like how you applied SLAP to improve the readability of your code. It looks like you are struggling with some parts of the code. Do take a look at some of your friends codes to get a better idea of how each functionality can be implemented and let me know if you require further assistance.

lunzard committed:
update Level 7: Save

improve the autoSave, autoLoad methods.

lunzard committed:
Fix problems of  Level-7: Save

lunzard committed:
create New classes + implement Ui Class

lunzard committed:
update A-MoreOOp

create Classes Storage, TaskList

lunzard committed:
update A-MoreOOP

implement Class Parser

lunzard committed:
Complete A-MoreOOP

lunzard committed:
Complete Level 8: Dates and Times

lunzard committed:
complete Level 9: Find

lunzard committed:
update A-JavaDoc

lunzard committed:
Merge pull request #1 from lunzard/branch-Level-8

Complete Level 8: Dates and Times

lunzard committed:
Merge pull request #2 from lunzard/branch-Level-9

complete Level 9: Find

lunzard committed:
Merge branch 'master' into branch-A-JavaDoc

lunzard committed:
Merge pull request #3 from lunzard/branch-A-JavaDoc

update A-JavaDoc

lunzard committed:
fix minor bugs

lunzard committed:
update A-UserGuide

lunzard committed:
update README

lunzard committed:
fix minor bugs

fix names of duke

lunzard committed:
fix minor bugs

remove static keywords

lunzard committed:
fix indentation

lunzard committed:
update A-JavaDoc

lunzard committed:
update ui
